python/lsst/ts/m2gui/display/figure.py

```diff
@@ -0,0 +1,153 @@
+# This file is part of ts_m2gui.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["Figure"]
+
+import numpy as np
+
+from PySide2.QtCore import Qt, QPointF
+from PySide2.QtCharts import QtCharts
+from PySide2.QtGui import QPainter
+
+
+class Figure(QtCharts.QChartView):
+    """Figure to show the line data.
+
+    This figure will allocate the memory space of the data points. The new
+    data will update the existed points on figure directly. This means if you
+    assign the "num_points" to be 10, you could not update the data to have >10
+    data points with self.update_data().
+
+    Parameters
+    ----------
+    range_x_min : `float` or `int`
+        Minimum of the range of x-axis.
+    range_x_max : `float` or `int`
+        Maximum of the range of x-axis.
+    num_points : `int`
+        Number of the points.
+    title_x : `str`
+        Title of the x axis.
+    title_y : `str`
+        Title of the y axis.
+    title_figure : `str`
+        Title of the figure.
+
+    Attributes
+    ----------
+    axis_x : `PySide2.QtCharts.QtCharts.QValueAxis`
+        X axis.
+    axis_y : `PySide2.QtCharts.QtCharts.QValueAxis`
+        Y axis.
+    """
+
+    # Minimum width of the chart's view
+    MINIMUM_WIDTH = 500
+
+    # Offset of the y axis
+    OFFSET_Y = 1
+
+    def __init__(
+        self, range_x_min, range_x_max, num_points, title_x, title_y, title_figure
+    ):
+        self._chart = QtCharts.QChart()
```

Comment:
QChartView provides chart() to access its chart. You are creating private value to store what was already stored, creating a maze.

The problem of course span from the fact you are mixing two things that are in Qt for good reason separated. It will be much better to plot your data in a subclass of QChart, and use QtCharts.QChartView for what it is intended, e.g. only chart display interface.

Reply:
Fixed in the commit: [af8c4ea](https://github.com/lsst-ts/ts_m2gui/pull/13/commits/af8c4eada9478d63f692c5cd50c53c057e3e2b34). Thanks!